packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1732,6 +1732,35 @@ abstract class MultiChildRenderObjectWidget extends RenderObjectWidget {
   /// If this list is going to be mutated, it is usually wise to put [Key]s on
   /// the widgets, so that the framework can match old configurations to new
   /// configurations and maintain the underlying render objects.
+  ///
+  /// Here's a common example where one mutates the children but forgets to set
+  /// a different key. In such case, the mutated children may not be reflected.
+  /// ```
+  /// class SomeWidgetState extends ... {
+  ///     List<Widget> _children;
+  ///
+  ///     void initState(...) {
+  ///         _children = [...];
+  ///     }
+  ///
+  ///     void someHandler() {
+  ///         setState(() {
+  ///             _children.add(...);
+  ///         });
+  ///     }
+  ///
+  ///     Widget build(...) {
+  ///         // A changing key is needed if someHandler modifies _children.
+  ///         return Row(children: _children/*, key: missing_key*/);
+  ///     }
+  /// }
+  /// ```
+  ///
+  /// Also, a [Widget] in Flutter is immutable, so directly modifying the
+  /// [children] such as `someMultiChildRenderObjectWidget.children.add(...)`
+  /// is a violation to Flutter's contract. Although no warning or error message
+  /// is printed out for such case right now, we advice against such usages to
+  /// avoid undefined behaviors.
```

Comment:
"advise against" is probably too weak, and "violation of Flutter's contract" is probably too strong-sounding. I would just say that passing the same actual List object each time will result in incorrect behaviour, and that a new List should be provided each time the class is built.